nixos/modules/programs/starship.nix

```diff
@@ -44,21 +44,27 @@ in
   config = mkIf cfg.enable {
     programs.bash.${initOption} = ''
       if [[ $TERM != "dumb" ]]; then
-        export STARSHIP_CONFIG=${settingsFile}
+        if [[ ! -f "$HOME/.config/starship.toml" ]]; then
```

Comment:
this looks weird, aint that an XDG path ? shouldn't it be using `${XDG_CONFIG_HOME:-.config}` ? is it possible to ask the starship executable instead what settings it loaded ?

Does  the home-manager module export STARSHIP_CONFIG ? in which case nixos could just set it conditionnally.

Reply:
You're probably right about the `XDG_CONFIG_HOME` - but the `starship` config documentation has this listed as a static path here: https://starship.rs/config/

I just went and double checked [the source code](https://github.com/starship/starship/blob/686bda1706e5b409129e6694639477a0f8a3f01b/src/configure.rs#L651) as well and I think it might actually be hard-coded to the home directory?

Home manager doesn't export `STARSHIP_CONFIG` in it's module; it just creates the config file at the right path.

How about a compromise?  I'll go and amend this to include a comment referencing the fact that starship appears to use a hardcoded config path - so our descendants will understand why this is a little weird?

Reply:
ty for checking, sounds about right. A comment should do the trick but please ping the starship maintainer I dont feel confident to merge something like this as an outsider.